src/systemd_ctypes/libsystemd.py

```diff
@@ -33,6 +33,25 @@
 from .typing import Annotated
 
 
+class Trampoline(ReferenceType):
+    deferred: 'ClassVar[list[Callback] | None]' = None
+    trampoline: Callback
+    userdata: UserData = None
+
+    def cancel(self) -> None:
+        self._unref()
+        self.value = None
+
+    def __del__(self) -> None:
+        # This might be the currently-dispatching callback — make sure we don't
+        # destroy the trampoline before we return.  We drop the deferred list
+        # from the event loop when we're sure we're not doing any dispatches.
+        if self.deferred is not None:
+            self.deferred.append(self.trampoline)
```

Comment:
This makes me a bit nervous -- `deferred` is a class variable and also set as such (on the class), but __del__ is an instance method. Can it happen that event sources or bus slots get freed at the same time in parallel threads? Does python do the locking here, or don't we support ctypes from multiple threads at all?

Also, could this be `Trampoiline.deferred`, to point out that this is a class var? Accessing it as `self` feels misleading (even though it probably does the same).

Reply:
It could definitely be accessed via the class.

Reply:
Done.

Reply:
I forgot about the GIL! So parallel access should be safe.